core-plugins/src/main/java/co/cask/hydrator/plugin/common/KinesisOutputFormat.java

```diff
@@ -0,0 +1,160 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.common;
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.kinesis.AmazonKinesisClient;
+import com.amazonaws.services.kinesis.model.ProvisionedThroughputExceededException;
+import com.amazonaws.services.kinesis.model.PutRecordRequest;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.OutputCommitter;
+import org.apache.hadoop.mapreduce.OutputFormat;
+import org.apache.hadoop.mapreduce.RecordWriter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/**
+ *
+ */
+public class KinesisOutputFormat extends OutputFormat {
+  private static final Logger LOG = LoggerFactory.getLogger(KinesisOutputFormat.class);
+  private static final int MAX_RETRIES = 3;
+  private static AmazonKinesisClient kinesisClient;
+
+  /**
+   * Get the {@link RecordWriter} for the given task.
+   *
+   * @param context the information about the current task.
+   * @return a {@link RecordWriter} to write the output for the job.
+   * @throws IOException
+   */
+  @Override
+  public RecordWriter getRecordWriter(TaskAttemptContext context) throws IOException, InterruptedException {
+    return new KinesisRecordWriter(kinesisClient, context);
+  }
+
+  /**
+   * Check for validity of the output-specification for the job and initialize {@link AmazonKinesisClient}
+   * <p>
+   * <p>This is to validate the output specification for the job when it is
+   * a job is submitted.</p>
+   *
+   * @param context information about the job
+   * @throws IOException when output should not be attempted
+   */
+  @Override
+  public void checkOutputSpecs(JobContext context) throws IOException, InterruptedException {
+    Configuration conf = context.getConfiguration();
+    BasicAWSCredentials awsCred = new BasicAWSCredentials(conf.get(Properties.KinesisRealtimeSink.ACCESS_ID),
+                                                          conf.get(Properties.KinesisRealtimeSink.ACCESS_KEY));
+
+    if (kinesisClient == null) {
+      LOG.info("Creating amazon kinesis client for stream {}", Properties.KinesisRealtimeSink.NAME);
+      kinesisClient = new AmazonKinesisClient(awsCred);
+      try {
+        KinesisUtil.createAndWaitForStream(kinesisClient, conf.get(Properties.KinesisRealtimeSink.NAME),
```

Comment:
This method simply returns if the stream already exists? If so, does the shard count take effect or no?